core/test/variable_operations_test.cpp

```diff
@@ -898,17 +935,17 @@ TEST(VariableProxy, scalar_operations) {
   auto var = makeVariable<double>({{Dim::Y, 2}, {Dim::X, 3}},
                                   {11, 12, 13, 21, 22, 23});
 
-  var.slice({Dim::X, 0}) += 1;
+  var.slice({Dim::X, 0}) += 1.0;
```

Comment:
Does it not work for `int` anymore? We should make sure it does, I guess?

Reply:
No, I did ask about this on Slack and we didn't come to any conclusion as to what is best.

Personally I prefer this method as I can't see many cases where you'd want to intentionally mix types.

Reply:
I think we will want to support a (selected) set of mixed operations. For example `double += float` makes sense, does `double += int`?

Reply:
So what do we actually want to support?

I can only think of:
- lhs int64, rhs int32
- lhs double, rhs float

I can't think of any reason to operate between integer and floating point.

Reply:
I agree that we should try and make the other combinations explicit, i.e., not support them for now.